### DISCUSSION TITLE
What's different between FFmpeg and VLC in waveform extraction?

### DISCUSSION THREAD

**Discussion by Mishasama** [SubtitleEdit/subtitleedit#5977] [Q&A] [answered]
It looks like the VLC is much faster?

**Comment by niksedk** [comment, marked as answer]
Actually, ffmpeg should be much faster - at least with a newer versions of ffmpeg (try to re-download ffmpeg from Options - Settings - Waveform - SE downloads ffmpeg 5.0.1 currently).

As a bonus, ffmpeg works for more formats than vlc.

The waveforms looks a little different, but I've not analyzed why...

**Reply by Mishasama** [reply]
> The waveforms looks a little different, but I've not analyzed why...

Maybe VLC output with audio preprocessing like below? Because it's a player, not just a decoder?

- VLC = File > Decode > Filters > Enhancements by Output Device (via Direct Sound or WASAPI) > **Enhanced Output** as result
- FFmpeg = File > Decode > **RAW Output** as result
